src/plugins/legacy/meshMapping/CMakeLists.txt

```diff
@@ -76,6 +75,7 @@ target_link_libraries(${TARGET_NAME}
   medCore
   medVtkInria
   medUtilities
+  ITKVtkGlue
```

Comment:
This should actually be `${VTK_LIBRARIES}` and `${ITK_LIBRARIES}` as the `find_package` handler for VTK and ITK fill in those variables with the proper libraries to link depending on which components where required.

Reply:
If it compiles without the necessary libraries being linked, it is because those libraries are being inherited from components which already did the link themselves (probably medVtkInria), except that it's not very clean to depend on that behaviour. Using ${VTK_LIBRARIES} and ${ITK_LIBRARIES} is the proper way to make sure there are no linking issues on any platform.

Reply:
(see https://cmake.org/cmake/help/v3.12/prop_tgt/INTERFACE_LINK_LIBRARIES.html#prop_tgt:INTERFACE_LINK_LIBRARIES)

Reply:
(actually this is clearer : https://cmake.org/cmake/help/v3.12/command/target_link_libraries.html#libraries-for-both-a-target-and-its-dependents)

Reply:
Thank you for your explanation to Mathilde, and thank you for the precious links.
It's very interesting.
I have just one question ${VTK_LIBRARIES} or ${ITK_LIBRARIES} contain all libraries or juste libraries defined by components modules list into the find_package?

Reply:
Normally it's just the ones needed for the required components. I tested this on my system and this is what it seems to do.

Reply:
(I also checked the VTK and ITK cmake code too and although it wasn't very clear it does seem to only take into account the required components)

Reply:
I'm fairly sure that I read something on this somewhere on a VTK or ITK doc, which is why I checked in the first place, but I don't remember where.

Reply:
Thanks

Reply:
No problem !

Reply:
If it's ok to use ${VTK_LIBRARIES} or ${ITK_LIBRARIES} as we did on music, it could be nice!

It will be easier to update our plugins. An other point is, as in this case, some of us (on linux mostly) do not have any problem compiling this code without ITKVtkGlue. So it's easier also, because i can't really know what module need to be added here.